src/pool/srv_pool.c

```diff
@@ -6777,3 +6811,88 @@ ds_pool_target_status_check(struct ds_pool *pool, uint32_t id, uint8_t matched_s
 
 	return target->ta_comp.co_status == matched_status ? 1 : 0;
 }
+
+int
+ds_pool_svc_load_map(struct ds_pool_svc *ds_svc, struct pool_map **map)
+{
+	struct pool_svc		*svc = pool_ds2svc(ds_svc);
+	struct rdb_tx		 tx = { 0 };
+	int			 rc;
+
+	rc = rdb_tx_begin(svc->ps_rsvc.s_db, svc->ps_rsvc.s_term, &tx);
+	if (rc == 0) {
+		ABT_rwlock_rdlock(svc->ps_lock);
+		rc = read_map(&tx, &svc->ps_root, map);
+		ABT_rwlock_unlock(svc->ps_lock);
+		rdb_tx_end(&tx);
+	}
+
+	if (rc != 0)
+		D_ERROR("Failed to load pool map for pool "DF_UUIDF": "DF_RC"\n",
+			DP_UUID(svc->ps_uuid), DP_RC(rc));
+
+	return rc;
+}
+
+int
+ds_pool_svc_flush_map(struct ds_pool_svc *ds_svc, struct pool_map *map, uint32_t version)
```

Comment:
[Question] Is it intentional that we do not schedule rebuild jobs when updating the pool map?

Reply:
The pool map update is driven by the DAOS check instead of regular rebuild. The logic is something like that:
For each target that reported as part of the pool, compare with the pool map and fix related inconsistency; and then handle those non-accessed (in former comparison) pool map entries. During these process, there will be yield because of RPC or interaction with admin. All related pool map fixes are in DRAM in this step. After all done, it will call ds_pool_svc_flush_map() to persistently change the pool map and broadcast the changes to other pool shards.